x-pack/legacy/plugins/maps/public/components/layer_addpanel/source_select/source_select.js

```diff
@@ -56,3 +65,15 @@ export function SourceSelect({
     </Fragment>
   );
 }
+
+function generateBetaBadge() {
+  return (
+    <EuiBetaBadge
+      label="Beta"
+      tooltipContent={i18n.translate('xpack.maps.sourceSelect.betaMessageBadge', {
+        defaultMessage:
+          'Maps GeoJSON Upload is still in beta. Please help us improve by reporting issues or bugs in the Kibana repo.',
```

Comment:
This message is GeojsonFileSource specific and should be supplied by GeojsonFileSource. Maybe add a property `GeojsonFileSource.betaMsg` and pull message from the source

Reply:
I agree that this function was too specific so I just made it more generic to handle any source title. I think in its current generic form, it makes more sense here